src/context.cpp

```diff
@@ -137,6 +146,56 @@ namespace librealsense
     }
 
 
+    template< class T >
+    static bool json_get_ex( json const & j, char const * key, T * pv )
+    {
+        auto it = j.find( key );
+        if( it == j.end()  ||  it->is_null() )
+            return false;
+        try
+        {
+            // This will throw for type mismatches, etc.
+            *pv = it->get< T >();
+        }
+        catch( std::exception const & e )
+        {
+            std::ostringstream s;
+            s << e.what() << " - while parsing '" << key << "'";
+            throw std::runtime_error( s.str() );
+        }
+        LOG_DEBUG( key << " = " << *pv );
+        return true;
+    }
+    template < class T >
+    static T json_get( json const & j, char const * key, T const & d )
+    {
+        T v;
+        if( json_get_ex< T >( j, key, &v ) )
+            return v;
+        LOG_DEBUG( key << " = " << d << " (default)" );
```

Comment:
If you didn't write (default) on the log string I wouldn't know d = default.
Maybe call it `default_val`? or add a comment before the signature?

Reply:
Done, `default_value`